lib/Dialect/Linalg/Transforms/LinalgTransforms.cpp

```diff
@@ -106,3 +111,38 @@ bool mlir::linalg::detail::isProducedByOpOfTypeImpl(
   }
   return false;
 }
+
+LogicalResult
+mlir::linalg::permuteGenericLinalgOp(PatternRewriter &rewriter, Operation *op,
+                                     ArrayRef<unsigned> permutation,
+                                     StringRef linalgMarker) {
+  // If permutation is empty, there is nothing to be done.
+  if (permutation.size() == 0)
+    return failure();
+
+  OpBuilder::InsertionGuard g(rewriter);
+  auto linOp = cast<LinalgOp>(op);
+  rewriter.setInsertionPoint(linOp);
```

Comment:
this shouldn't be necessary, the rewrite pattern should set you up in the proper location already

Reply:
Removed. Thanks.